xtra-libp2p-rollover/src/maker.rs

```diff
@@ -137,50 +149,61 @@ where
-    }
+        if !self.is_accepting_rollovers {
+            emit_rejected(order_id, &self.executor).await;
 
-    async fn handle(&mut self, msg: Accept) -> Result<()> {
-        let Accept {
-            order_id,
-            tx_fee_rate,
-            long_funding_rate,
-            short_funding_rate,
-        } = msg;
+            let mut tasks = Tasks::default();
+            tasks.add_fallible(
+                async move {
+                    framed
+                        .send(ListenerMessage::Decision(Decision::Reject(Reject {
+                            order_id,
+                        })))
+                        .await
+                },
+                move |e| async move {
+                    tracing::debug!(%order_id, "Failed to send reject rollover to the taker: {e:#}")
+                },
+            );
+            self.protocol_tasks.insert(order_id, tasks);
+
+            return;
+        }
 
         fn next_rollover_span(parent: &tracing::Span) -> tracing::Span {
             tracing::debug_span!(parent: parent, "next rollover message")
         }
 
-        let (mut framed, _, from_params) =
-            self.pending_protocols.remove(&order_id).with_context(|| {
-                format!("No active protocol for {order_id} when accepting rollover")
-            })?;
-
         let mut tasks = Tasks::default();
         tasks.add_fallible(
             {
                 let executor = self.executor.clone();
                 let oracle = self.oracle.clone();
+                let rates =  self.rates.clone();
                 let oracle_pk = self.oracle_pk;
                 let n_payouts = self.n_payouts;
                 async move {
+                    let Rates {
+                        funding_rate_long,
+                        funding_rate_short,
+                        tx_fee_rate,
+                    } = rates
+                        .get_rates()
+                        .await
+                        .context("Failed to get rates")?;
+
                     let (rollover_params, dlc, position, oracle_event_id, funding_rate) = executor
                         .execute(order_id, |cfd| {
                             let funding_rate = match cfd.position() {
-                                Position::Long => long_funding_rate,
-                                Position::Short => short_funding_rate,
+                                Position::Long => funding_rate_long,
+                                Position::Short => funding_rate_short,
                             };
 
                             let (event, params, dlc, position, oracle_event_id) = cfd
                                 .accept_rollover_proposal(
                                     tx_fee_rate,
                                     funding_rate,
-                                    Some(from_params),
+                                    Some((from_event_id, from_complete_fee)),
```

Comment:
Something in this commit seems odd, `from_event_id` and `complete_fee` are removed, but here we are still using it - where are they coming from? I assume the changes are not cleanly applied after a rebase.

Reply:
My bad